.cirrus.yml

```diff
@@ -0,0 +1,272 @@
+    cpu: 2
+    memory: 2G
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  setup_script:
+    - source cirrus-env BUILD # Generates a build number
+    - echo $BUILD_NUMBER
+    - export PACKAGE_VERSION=$(npm pkg get version | tr -d \" | sed "s|-SNAPSHOT|+$BUILD_NUMBER|g")
+    - echo PACKAGE_VERSION=${PACKAGE_VERSION}
+
+deploy_universal_vsix_task:
+  depends_on: build
+  <<: *SKIP_MASTER_BRANCH_AND_ALLVSIX_BRANCH
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+    cpu: 2
+    memory: 2G
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  deploy_script:
+    - source cirrus-env QA # Retrieve build number
+    - node_modules/.bin/gulp deploy
+
+deploy_all_vsix_files_task:
+  <<: *ONLY_MASTER_BUILD_OR_ALLVSIX_BRANCH
+  depends_on: build
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+    cpu: 2
+    memory: 2G
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  deploy_all_script:
+    - source cirrus-env QA # Retrieve build number
+    - node_modules/.bin/gulp deploy-all
+
+test_windows_task:
+  # name: Run tests on Windows
+  depends_on:
+    - build
+    - deploy_all_vsix_files
+    - deploy_universal_vsix
+  <<: *ONLY_IF_EXCEPT_NIGHTLY
+  ec2_instance:
+    <<: *WINVM_DEFINITION
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  env:
+    NODE_ENV: 'continuous-integration'
+  script:
+    - source cirrus-env QA
+    - export packageVersion=$(npm pkg get version | tr -d \")
+    - npm run compile
+    - npm run prepare
+    - npm run test-cov
+  always:
+    unit_tests_artifacts:
+      path: '**/alltests.xml'
+      format: junit
+
+test_linux_task:
+  #  name: Run tests, Sonarqube analysis on Linux
+  depends_on:
+    - build
+    - deploy_all_vsix_files
+    - deploy_universal_vsix
+  <<: *ONLY_IF_EXCEPT_NIGHTLY
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+    cpu: 4
+    memory: 4G
+  env:
+    DISPLAY: ":10"
+    SONARQUBE_NEXT_TOKEN: VAULT[development/kv/data/next data.token]
+    SONARQUBE_NEXT_URL: VAULT[development/kv/data/next data.url]
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  start_xvfb_background_script:
+    - /etc/init.d/xvfb start
+  script:
+    - sleep 5
+    - source cirrus-env QA
+    - export packageVersion=$(npm pkg get version | tr -d \")
+    - npm run compile
+    - npm run prepare
+    - npm run test-cov
+    - .cirrus/cirrus_sonar_analysis_npm
+
+  always:
+    unit_tests_artifacts:
+      path: '**/alltests.xml'
+      format: junit
+
+mend_scan_task:
+  depends_on:
+    - test_linux
+  <<: *ONLY_MAIN_BRANCHES
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  env:
+    WS_APIKEY: VAULT[development/kv/data/mend data.apikey]
+    JDK_VERSION: "11"
+  whitesource_script:
+    - source cirrus-env QA # read build number
+    - export PROJECT_VERSION=$(npm pkg get version | tr -d \" | sed "s|-SNAPSHOT||g")
+    - source ws_scan.sh
+  allow_failures: "true"
+  always:
+    ws_artifacts:
+      path: "whitesource/**/*"
+
+qa_task:
+  #  name: Run ITs
+  depends_on:
+    - test_linux
+    - test_windows
+    - mend_scan
```

Comment:
I believe this is not correct: the QA only depends on the availability of the universal VSIX.

I think that removing the dependency on the `test_` tasks would increase parallelism (and thus overall pipeline time).

Reply:
No, this is not what is expected. The QA needs the universal VSIX to be built, either by the dedicated task, or by the task that generates all VSIXes.

Builds on `master` must build and promote all VSIX packages, not only the platform-specific ones.